engine/kv_engine.cpp

```diff
@@ -1272,6 +1285,7 @@ Status KVEngine::StringDeleteImpl(const StringView &key) {
     case Status::Ok:
       assert(entry_ptr->header.status == HashEntryStatus::Updating);
       purgeAndFree(hash_entry.index.string_record);
+      TEST_SYNC_POINT("DataEntry::StringDeleteImpl::purgeAndFree");
```

Comment:
"StringDeleteImpl::AfterDeletion"
The spin is guarded. There's no way other thread can interrupt next step.